src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -467,8 +467,9 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 Host.AssertValue(input);
 
                 var activeOutputColNames = _parent.Outputs.Where((x, i) => activeOutput(i)).ToArray();
+                var activeIndex = _parent.Model.ModelInfo.OutputNames.IndexOf(_parent.Outputs[iinfo]);
```

Comment:
The reason for this may not be obvious when someone is reading the code. Can you please add a small note about why this has been added?

Reply:
Yeah, I think it's somewhat confusing what's happening here. But instead of adding a note I'd suggest the following:
1. Add a new method or property to OnnxTransformer called "GetOutputInfo(int iinfo)", and comment it saying that it maps the "iinfo" which is the index related to ML.NET DataView to return the corresponding OutputInfo of the OnnxModel.
2. Let GetOutputInfo return `Model.ModelInfo.OutputsInfo[Model.ModelInfo.OutputNames.IndexOf(Outputs[iinfo])`
3. Then here instead of using `_parent.Model.ModelInfo.OutputsInfo[activeIndex]` it would simply be `_parent.GetOutputInfo(iinfo)`.

This way there's no need to introduce the new "activeIndex" variable. By taking a quick look at the code, it does seem doable, and also I think it's convenient, because `Model.ModelInfo.OutputsInfo[]` is actually only used in 3 places, where Michael has already put the `activeIndex` variable (i.e. whenever we use that array, we must make sure to map correctly the index as Michael did in here).

---
In reply to: [434145568](https://github.com/dotnet/machinelearning/pull/5192#discussion_r434145568) [](ancestors = 434145568)

Reply:
Is the overhead of a new method worth it for just that one line replacement? This isn't in the critical path so probably doesn't matter too much, just wanted your thoughts. I'm fine either way.

Reply:
If something can be better expressed through code instead of comments, I would lean towards expressing it as code. However, I don't think `GetOutputInfo(int iinfo)` fully conveys that meaning either that is mapping the dataview column to the onnx model output variable. Maybe something like `MapDataViewColumnToOnnxOutputVariable(int iinfo)` (or something similar if you can find a shorter but equivalent name). 

But you may still need to add some comments to explain why this mapping is necessary. 

So, I am okay however you choose to fix this as long as the context and change are explained through some combination of code and comments.

---
In reply to: [434186456](https://github.com/dotnet/machinelearning/pull/5192#discussion_r434186456) [](ancestors = 434186456)

Reply:
That is a really good point Harish. I will make a new method named something like your suggestion, and make sure that there are enough comments for it.

Reply:
Alright, the method and comments are added. I still left the local variable since the method name is long, but if you guys prefer I can remove it and just use the method directly.